test/Microsoft.Azure.WebJobs.Host.EndToEndTests/ApplicationInsights/ApplicationInsightsEndToEndTests.cs

```diff
@@ -321,10 +326,57 @@ public async Task ApplicationInsights_OuterRequestIsTrackedOnce()
                 RequestTelemetry functionRequest = requestTelemetries.Single();
                 Assert.Same(outerRequest, functionRequest);
 
+                Assert.True(double.TryParse(functionRequest.Properties[LogConstants.FunctionExecutionTimeKey], out double functionDuration));
+                Assert.True(functionRequest.Duration.TotalMilliseconds >= functionDuration);
+
                 ValidateRequest(functionRequest, testName, true);
             }
         }
 
+        [Fact]
+        public async Task ApplicationInsights_HttpRequestTracking()
+        {
+            var fakeAspNetCore = new FakeAspNetCore();
+            string testName = nameof(TestApplicationInsightsInformation);
+            using (IHost host = ConfigureHost())
+            {
+                await host.StartAsync();
+
+                var httpContext = fakeAspNetCore.CreateFakeContext(new Dictionary<string, string>
+                {
+                    ["traceparent"] = "00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01"
+                });
+
+                // simulate auto tracked HTTP incoming call
+                var activity = fakeAspNetCore.StartRequest(httpContext);
+
+                MethodInfo methodInfo = GetType().GetMethod(testName, BindingFlags.Public | BindingFlags.Static);
+                await host.GetJobHost().CallAsync(methodInfo, new { input = "input" });
+                fakeAspNetCore.StopRequest(activity, httpContext);
```

Comment:
One thing I'm not following fully -- It seems like our function Logger scope that sets the Category would have fallen out of scope by the time the RequestTelemetry is logged by App Insights. How is it getting that correct Category of Host.Results?

Reply:
The timeline is

0. AspNetCore starts a new Activity
1. AI starts tracking request (based on AspNetCore events). 
2. Function execution starts and sets function scope
3. Function execution ends and ApplicationInsightsLogger copies category to the Activity.Tags
4. AspNetCore sends Stop event and ApplicationInsights tracks Request
5. WebJobsTelemetryIntitializer copies Activity.Tags to Request.Properties
6. AspNEtCore stops Activity
7. FilteringProcessor checks category and lets request telemetry through as it has category

Reply:
Ah ha yes! Step 3 was the one I was missing somehow.